game-core/src/main/java/games/strategy/engine/framework/startup/login/HmacSha512Authenticator.java

```diff
@@ -41,31 +41,36 @@ private HmacSha512Authenticator() {}
    *         node; never {@code null}.
    */
   static Map<String, String> newChallenge() {
+    // This length is relatively large for a nonce simply because we do not currently expire nonces after they have been
+    // issued. If this were done, and a short nonce lifetime was used, we could probably get away with 8 bytes of random
+    // data safely as long as we included the timestamp as part of the nonce.
+    //
+    // https://security.stackexchange.com/questions/1952/how-long-should-a-random-nonce-be
+    final int nonceLengthInBytes = 64;
+
+    // Salt length should match hash output size.
+    //
+    // https://crackstation.net/hashing-security.htm#salt
+    final int saltLengthInBytes = 64;
+
     return Maps.newHashMap(ImmutableMap.<String, String>builder()
-        .put(encodeProperty(ChallengePropertyNames.NONCE, newRandomBytes()))
-        .put(encodeProperty(ChallengePropertyNames.SALT, newRandomBytes()))
+        .put(encodeProperty(ChallengePropertyNames.NONCE, newRandomBytes(nonceLengthInBytes)))
+        .put(encodeProperty(ChallengePropertyNames.SALT, newRandomBytes(saltLengthInBytes)))
         .build());
   }
 
   private static Map.Entry<String, String> encodeProperty(final String name, final byte[] value) {
     return Maps.immutableEntry(name, Base64.getEncoder().encodeToString(value));
   }
 
-  private static byte[] newRandomBytes() {
+  private static byte[] newRandomBytes(final int length) {
```

Comment:
The constant-value-parameter analysis is smart enough to detect when two constants are different but have the same value. If the two values are just local values, then the analysis flags the unnecessary method parameter.

To really fix things, we'd just need to make the nonce and salt lengths proper constants.